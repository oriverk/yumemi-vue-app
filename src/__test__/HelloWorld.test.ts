import { mount } from "@vue/test-utils";
import HelloWorld from "../components/HelloWorld.vue";

describe("HelloWorld", () => {
  test("hello test", async () => {
    console.log(HelloWorld.__name)
    expect("HelloWorld").toBe(HelloWorld.__name)
  })
  

  test('mount component', async () => {
    expect(HelloWorld).toBeTruthy();

    const wrapper = mount(HelloWorld, {
      props: {
        msg: "hello test"
      }
    })

    expect(wrapper.text()).toContain('hello test')
    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.get('button').trigger('click')
    expect(wrapper.text()).toContain('count is 1')

    await wrapper.get('button').trigger('click')
    expect(wrapper.text()).toContain('count is 2')
    
  })
})