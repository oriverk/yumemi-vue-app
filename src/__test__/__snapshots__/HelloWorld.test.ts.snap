// Vitest Snapshot v1

exports[`HelloWorld > mount component 1`] = `
"<h1 data-v-e17ea971=\\"\\">hello test</h1>
<div class=\\"card\\" data-v-e17ea971=\\"\\"><button type=\\"button\\" data-v-e17ea971=\\"\\">count is 0</button></div>
<p class=\\"read-the-docs\\" data-v-e17ea971=\\"\\">Click on the Vite and Vue logos to learn more</p>"
`;
